### DISCUSSION TITLE
weekly exercise ran twice today: once each in two different time zones

### DISCUSSION THREAD

**Discussion by billpribyl** [jgyates/genmon#1141] [Questions] [answered]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).

### Version

1.19.03

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have submitted logs

### What is/are your question(s)?

My weekly exercise time is set to Thursdays at 18.30. My generator insists on setting its time zone to UCT, so it launched the weekly exercise today at 18.30 UCT, which is 13.30 in my local time zone, which is America/Chicago (DST).  Then, at 18.30 in my local time zone, the exercise ran for the second time today.

I've confirmed that my Pi time zone is set to America/Chicago. I am not using the enhanced exercise add-on in genmon.

Genmon Maintenance page shows:
   Generator Exercise Time: Weekly Thursday, 18:30

Genmon status page currently shows:
   Time :
        Monitor Time : Thursday June 27, 2024 19:43:11
        Generator Time : Friday June 28, 2024 00:43

Easy fix? 

(I would love for my Evolution 2 controller to have the correct time zone but thus far I have not located the secret pathway to make that happen.)

**Comment by jgyates** [comment]
Do you have "Sync Time" enabled on the setting page? If enabled this will sync the generator time to the pi time once an hour.

Do you have Generac Mobile Link connected at the same time as genmon? Is the time zone displayed some where? I am not aware of a time zone setting in the controller so I am guessing this is being set by mobile link?

**Comment by billpribyl** [comment, marked as answer]
Yes, Sync Tmie is enabled. Yes, I have Mobile Link connected. I don't see
any way to set the time zone in the app.

I will disable Sync Time and learn to live in UCT. Thanks.

**Comment by billpribyl** [comment]
Separately, I was having a lot of timeout errors, and the Mobile Link app thought the generator was Off but it was on Auto. At the suggestion of pintsize.me, I cycled the power on the Evolution 2 controller by disconnecting the power connector and removing the fuse. When the controller came back up, it eventually set itself (correctly) to America/Chicago time zone...and Mobile Link reconnected properly.

Including this detail in case someone else experiences time zone or Mobile Link connection issues.

**Comment by billpribyl** [comment]
Cycling power probably had nothing to do with it. I just saw the following messages in the Mobile Link app:

07/02/2024 at 12:54 PM
Exercise Time Issue
We have found an issue causing some generator models to not correctly verify your current time zone via an online connection. This is causing the affected generators to default to "UTC" or 0 time zone causing possible unexpected exercise schedules.
Please note there is nothing you need to do currently as this does not affect the performance of your generator.
Our team is aware of the time zone issue and currently working on a fix with an estimated release by July 5th.
We apologize for any inconvenience this may cause and appreciate your patience as we work to resolve this matter.


07/03/2024 at 3:16 PM
Exercise Time Issue Resolved
We have resolved the issue causing some generator models to not correctly verify the current time zone via an online connection.
If you were impacted by this issue, it has been resolved.
We have been able to remotely update the time zone of any affected generators.
To ensure that you do not experience the issue again please update your Mobile Link application to version 3.1.2 or higher or turn on auto app updates in your app store, to always stay up to date.

**Reply by jgyates** [reply]
Thanks for the info.
